core/src/col_metadata.rs

```diff
@@ -171,9 +171,12 @@ pub struct VersionedJsonSchema {
 }
 
 // Struct representing the ResultSetSchema.
+// The `schema` field needs the alias `result_set_schema` because this struct is used to get the schema
+// from the __sql_schemas collection, which stores the schema in it's `schema` field, and the libmongosqltranslate
+// `translate` command, which stores the schema in it's `result_set_schema` field.
 #[derive(Serialize, Deserialize, PartialEq, Debug, Clone, Default)]
 pub struct ResultSetSchema {
-    #[serde(rename = "result_set_schema")]
+    #[serde(alias = "result_set_schema")]
     pub schema: crate::json_schema::Schema,
```

Comment:
I'm not sure why this wasn't failing earlier, but I fixed it and added a comment explaining what's going on.

Reply:
Oh actually I think it's because the error it would cause gets pushed into a `warnings` vector instead of causing a crash. This is probably why my tests weren't failing.

```
let result_set_schema: Result<ResultSetSchema> =
                                ResultSetSchema::from_sql_schemas_document(&schema_doc).map_err(
                                    |e| {
                                        Error::CollectionDeserialization(collection_name.clone(), e)
                                    },
                                );

                            match result_set_schema {
                                Ok(result_set_schema) => result_set_schema,
                                Err(error) => {
                                    // If there is an Error while deserializing the schema, we won't show any columns for it
                                    warnings.push(error);
                                    continue;
                                }
                            }
```

Reply:
Oh wait never mind. The part of the code that would fail was never called by my tests. That's why this wasn't causing any failures.